station.py

```diff
@@ -192,16 +192,11 @@ def prepare_image(client):
     file_conv = IMAGE_WORKDIR + bytes(client).hex().upper() + "_" + str(imgVer) + ".bmp" # also use the MAC in case 1 images are created within 1 second
 
     if not os.path.isfile(file_conv):
-        if is_bmp == 1:
-            bmp2grays.convertImage(1, "1bppR", filename, file_conv)
-        else:
-            pf = open(filename,mode='rb')
-            imgData = pf.read()
-            pf.close()
-            pngdata = BytesIO(imgData)
-            im = Image.open(pngdata)
-            im_L = im.convert("1")
-            im_L.save(file_conv)
+        if is_bmp == 0:
+            Image.open(bytes(client).hex() + ".png").convert("RGB").save(bytes(client).hex() + ".bmp")
+        bmp2grays.convertImage(1, "1bppR", bytes(client).hex() + ".bmp", file_conv)
```

Comment:
Are you overwriting the existing bmp file here during the conversion?

Reply:
Yes and no. 
It will overwrite any MAC.bmp file but since it is using the .png file in that moment it should not be bad.

But a better solution really is to just use some temp /tmp/random.bmp file for that.

Will change it

Reply:
But if you provide MAC.bmp, it will overwrite that one, correct? That is my concern.